src/ekam/ekam.cpp

```diff
@@ -431,6 +433,28 @@ void scanSourceTree(File* src, Driver* driver) {
   }
 }
 
+OwnedPtr<Dashboard> getDashboard(int maxDisplayedLogLines) {
+  if (!isatty(STDOUT_FILENO)) {
+    return newOwned<SimpleDashboard>(stdout);
+  }
+
+  // Sanity check: make sure the window size is non-zero; otherwise something
+  // is messed up about the terminal, and we should assume it doesn't work
+  // correctly.
+  //
+  // See issue #29
+  struct winsize windowSize;
+  ioctl(STDOUT_FILENO, TIOCGWINSZ, &windowSize);
```

Comment:
Probably should check for errors on this ioctl() call? (Could also fall back to SimpleDashboard in that case.)